### PR TITLE
refactor: move none biz code to utils

### PR DESCRIPTION
**Pull request by genffy**
- add crypto openapi user utils
- remove duplicate user and secret check

### CODE REVIEW

File: src/api/chat.ts [hunk 1]
```diff
@@ -50,3 +50,20 @@ export async function sendMessage(conversageId: number, message: string, name?:
...
+      messages: [
+        {
+          role: "user",
+          content: message,
+          name: name ?? undefined,
+        },
+      ],
+    }),
+  });
+}
```
Comment (CGQAQ) [inline src/api/chat.ts:69]: this

File: src/pages/api/chatgpt/chat.ts [hunk 2]
```diff
@@ -101,6 +55,18 @@ const handler: NextApiHandler = async (req, res) => {
...
+          return sendMsgsUseStream({
+            res,
+            client: chatClient,
+            conversation_id: conversation_id,
+            msgs: chats.map(
+              (it) => ({ role: it.role, content: it.content, name: it.name } as ChatCompletionRequestMessage),
+            ),
+            newMsgs: body.messages,
+          });
+        }
```
Comment (CGQAQ) [inline src/pages/api/chatgpt/chat.ts:69]: this

File: src/pages/api/chatgpt/chat.ts [hunk 3]
```diff
@@ -186,3 +152,78 @@ async function sendMsgs({
     res.status(500).json({ error: msg });
   }
 }
+// WIP: not real stream
+// refs:
+// https://github.com/vercel/next.js/issues/9965#issuecomment-614823642
+// https://beta.nextjs.org/docs/routing/route-handlers#streaming
+async function sendMsgsUseStream({
```
Comment (CGQAQ) [inline src/pages/api/chatgpt/chat.ts:159]: this

File: tsconfig.json [hunk 4]
```diff
@@ -39,10 +39,9 @@
     "next-env.d.ts",
     "**/*.ts",
     "**/*.tsx",
-    ".next/types/**/*.ts",
-    "/Volumes/source/ai/prompt-cheatsheet/.next/types/**/*.ts"
+    ".next/types/**/*.ts"
   ],
```
Comment (CGQAQ) [inline tsconfig.json:43]: good catch

**Review by CGQAQ — COMMENTED** [review]
I saw you have several WIPs, do you want to finish these OR merge early?

**Comment by genffy** [comment]
> I saw you have several WIPs, do you want to finish these OR merge early?

revert WIP commit, move to another draft pr

**Comment by CGQAQ** [comment]
LGTM, thanks

**Pull request merged**